### PR TITLE
Added a new ADRs section.

### LINKED ISSUE #181: Add support for ADRs
ADRs are Architecture Decision Records; having them alongside our Engineering Handbook is useful. Usually, we see ADRs used within a project setting to document architecture or decision-related decisions made by the team. At the handbook level, we can use it to document decisions made by the WSM and technical leadership that apply to most projects we work on at Axelerant.

General links:
- https://adr.github.io/
- https://github.com/joelparkerhenderson/architecture-decision-record

This issue is to build support for ADRs within our handbook. We need a new archetype for this along with the existing one (`docs`). This means our ADRs will be placed in `content/adrs` directory. Each new ADR will be proposed as a new markdown file within this directory. Like all ADRs, each file should support some metadata such as:
- category
- date
- status (proposed, accepted, declined, deprecated)
- decision

The entire markdown content would explain the ADR anyway but the metadata can be used for filtering and generating lists. The metadata can live within the YAML frontmatter of the markdown file.

In this issue, let's limit our tasks to the following:

- [ ] Support for a new archetype for ADRs
- [ ] Placing a link to the main ADR listing page in a prominent section in the handbook layout
- [ ] List all the accepted ADRs sorted by the date and show the category and decision

We can worry about improving this with filters, sorts, etc, in follow-up issues.

Reference:
- Reference site: https://architecture.lullabot.com/adrs/
- Gatsby theme for ADRs: https://github.com/Lullabot/gatsby-theme-adr
- Example ADR for use with the above theme: https://raw.githubusercontent.com/Lullabot/gatsby-theme-adr/main/packages/example/adrs/20210608-record-architecture-decisions.mdx

### PR DESCRIPTION
**Pull request by PGaigole**
Fixes #181

### CODE REVIEW

File: assets/_custom.scss [hunk 1]
```diff
@@ -744,3 +744,59 @@ cite.bq__cite {
     color: $gray;
   }
 }
+
+// ADR theming
+.adr-card {
```
Comment (hussainweb) [inline assets/_custom.scss:749]: How about creating a new SCSS file to give some separation?

File: content/docs/adrs/angular.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+---
+title: 'Angular'
+weight: 2
+---
+
+# ADR for Angular
```
Comment (hussainweb) [inline content/docs/adrs/angular.md:6]: Is this supposed to be a single ADR? It doesn't look that way.

My vision is that we create a new file for each ADR and write things in a common markdown. In the frontmatter, we put in some metadata related to category, decision, status, etc. This could be used in listings and even filters (but that's something we can figure out later). For now, we should just be able to list all the ADRs (each ADR is a separate markdown file) and sort by date rather than weight. Pages such as "Angular" and "Drupal" (assuming they are actually category of ADRs, should be rendered automatically.

**Review by hussainweb — CHANGES_REQUESTED** [review]

File: content/adr/angular-fe.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+status: "proposed"
+categories:
+  - Angular
+---
+
+# ADR for Angular
+
+Below is a list of all Architecture Decision Records (ADRs) for our Angular project.
+
+{{< adr-list >}}
```
Comment (hussainweb) [inline content/adr/angular-fe.md:15]: We shouldn't have this in each ADR file. As such, we don't even need a listing on each ADR. The list should only be there on the main ADR page.

File: assets/_custom.scss (hunk 1 above)
Comment (hussainweb) [reply assets/_custom.scss:749]: I see the new file but this still needs to be removed.

File: content/docs/how-we-work/coding-principles-standards.md [hunk 4]
```diff
@@ -46,6 +46,30 @@ In our projects, we strictly adhere to the established coding standards. These s
 
   - When writing code, our code should solve ONLY and EXACTLY the use case that we are trying to support. In many instances, we have a desire to make things MORE FLEXIBLE in the FUTURE but that is code we will have to support without knowing whether any of those use cases will ever come up. Another big reason for wanting to be FLEXIBLE is that refactoring is hard, and it feels easier to have flexibility at the front, but that is why we write automated tests, so refactoring will not be as painful.
 
+## Quality as Teamwork
```
Comment (hussainweb) [inline content/docs/how-we-work/coding-principles-standards.md:49]: This is unrelated. Maybe you need to rebase once?

Comment (PGaigole) [reply content/docs/how-we-work/coding-principles-standards.md:49]: I did rebase this. I'll try it again. Thanks!

**Review by hussainweb — CHANGES_REQUESTED** [review]
Can you have an example ADR with detailed markdown to see how it renders? Right now, all the markdown files are just metadata.

Also, please remove the ADR relevant styles from _custom.scss.

File: content/adr/_index.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+An ADR typically includes the following sections:
+
+1. **Title**: A clear and concise title for the decision.
+2. **Date**: The date when the decision was made.
+3. **Status**: The current status of the decision (e.g., proposed, accepted, deprecated).
+4. **Context**: A brief explanation of the situation that led to the need for the decision.
+5. **Decision**: The decision that was made.
+6. **Consequences**: The outcomes and consequences of the decision.
+7. **Alternatives Considered**: Other options that were considered and why they were not chosen.
```
[the remark this answers is not in the document]
Comment (hussainweb) [reply content/adr/_index.md:34]: @Kalaiselvan88, let's keep the tags for a follow-up issue. It's not a challenge to add it as Tags are already a built-in taxonomy. But please create a follow-up issue to display the tag and create listings, etc.

File: content/adr/angular-fe.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
+---
+title: 'Adopt Angular for Frontend Development'
+type: "adrs"
```
Comment (hussainweb) [inline content/adr/angular-fe.md:3]: What does `type` do?

File: content/adr/angular-fe.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+date: "05-07-2024"
+decision: "We will adopt Angular as our frontend framework due to its comprehensive tooling, strong typing with TypeScript, and suitability for building large-scale applications."
+status: "proposed"
+categories:
+  - Angular
```
Comment (hussainweb) [inline content/adr/angular-fe.md:8]: We should add these to the `archetypes/adr.md` with default values.

File: archetypes/adrs.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (hussainweb) [inline archetypes/adrs.md:1]: I just checked and this file has no effect. Running `hugo new content adr/something.md` doesn't utilize this template. It just looks like it does because it matches the default template.

We have to rename this to `adr.md` for it to take effect. Also, we should add more fields like `decision`, `status`, and `categories` to this template.

**Review by hussainweb — CHANGES_REQUESTED** [review]

File: content/adr/angular-fe.md (hunk 6 above)
Comment (PGaigole) [reply content/adr/angular-fe.md:3]: The 'type' here would help us to get all ADRs and if we haven't created any ADRs then we could create a fallback condition like below - 
{{ $adrPages := where $.Site.RegularPages "Type" "adrs" }}

Comment (hussainweb) [reply content/adr/angular-fe.md:3]: Hmm... Is this a result of the earlier code where ADRs were of `docs` archetype? I am wondering why this is needed because we should be able to just look up all the content of archetype `adr`.

File: archetypes/adrs.md [hunk 9]
```diff
@@ -0,0 +1,5 @@
...
+title: "{{ replace .Name "-" " " | title }}"
+date: {{ .Date }}
```
Comment (hussainweb) [inline archetypes/adrs.md:3]:
```suggestion
date: '{{ time.Now.Format "2006-01-02" }}'
```

File: archetypes/adrs.md (hunk 8 above)
Comment (hussainweb) [reply archetypes/adrs.md:1]: Further, we should also include template content here. It is supported as per https://gohugo.io/content-management/archetypes/#include-content

Comment (PGaigole) [reply archetypes/adrs.md:1]: Yes, you are right. I will make these changes accordingly.

File: content/adr/angular-fe.md (hunk 6 above)
Comment (PGaigole) [reply content/adr/angular-fe.md:3]: @hussainweb I will go ahead remove the 'type' then for now and find a better way to list out all ADRs.

File: archetypes/adr.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
+---
+title: "{{ replace .Name "-" " " | title }}"
+type: "adrs"
```
Comment (hussainweb) [inline archetypes/adr.md:3]: I think you forgot to remove this based on your comment.

File: archetypes/adr.md [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+date: "{{ time.Now.Format "2006-01-02" }}"
+decision: "Decision summary goes here."
+status: "proposed"
```
Comment (hussainweb) [inline archetypes/adr.md:6]: I am wondering if we should have a `draft` status. Do you see value in adding that?

@PGaigole, What are the different values for status as per you?

File: archetypes/adr.md (hunk 10 above)
Comment (PGaigole) [reply archetypes/adr.md]: Added now.

File: archetypes/adr.md (hunk 11 above)
Comment (PGaigole) [reply archetypes/adr.md:5]: @hussainweb I had three values in mind. Accepted, Proposed and Declined.
I did see the documentation had a 4th value as Deprecated

Comment (hussainweb) [reply archetypes/adr.md:5]: Hmm... I think draft might be useful in case the ADR is still not ready for review. But we can support that later. Setting it to "proposed" by default is good for now.

**Review by hussainweb — APPROVED** [review]
Let's merge this and go ahead with all the follow-ups we discussed.
